### DISCUSSION TITLE
Allow multiple input targets (comma separated globs, array of inputs)  (👍44)

### DISCUSSION THREAD

**Discussion by ghost** [11ty/eleventy#3996] [Enhancement Queue]
It would be nice to have a feature to pass multiple elements (directories, files, globs) to the `dir.input` option, such as:

```js
// .eleventy.js
module.exports = function eleventyConfiguration(config) {
    return {
        dir: {
            input: ['src/*.md', 'anotherSrc/*.njk'],
            output: 'dist',
        },
        ...
    };
};
```

because something like this doesn't work:

```js
// .eleventy.js
module.exports = function eleventyConfiguration(config) {
    return {
        dir: {
            input: '{src/*.md,anotherSrc/*.njk}',
            output: 'dist',
        },
        ...
    };
};
```

**Comment by zachleat** [comment]
I like this idea too!

> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by AlexNodex** [comment]
Throwing another possible option to the mix, this would allow for good flexibility.

```js
module.exports = function(eleventyConfig) {
  return {
    dir: {
      input: function() {
        const directories = aFunctionToReturnInputPathsBasedOnAConfigFile();
        return directories;
      },
      output: "public"
    }
  };
};
```

In my opinion one of: arrays, globs, a function and a string would be the best approach for the most flexible input

**Comment by edwardhorsford** [comment]
I've just come here looking for similar. I was looking to explore splitting my current `views` folder in to separate `pages` and `posts` folders.

**Comment by Meeshalk** [comment]
I wanted to have `pages` folder with multiple `.html` pages. Seems like its not possible ay the moment.

**Comment by noveogroup-amorgunov** [comment]
Hey! I would like to use same feature to split blog posts and 11ty app to diff monorepo packages like:

```ts
root
  packages
    app // 11ty app
    content
      posts
        2025-05-20-test.md
        2025-05-21-test-2.md
```

UPDATE NOTES:

if someone faces the same problem, I solved it through coping posts from content package to app with [`npm-watch`](https://www.npmjs.com/package/npm-watch):

```
// package.json

  "watch": {
    "copy:posts": {
      "patterns": ["../content/posts"],
      "extensions": "md"
    }
  },
  "scripts": {
    "prestart": "pnpm run \"/^copy:.*/\"",
    "prebuild": "pnpm run \"/^copy:.*/\"",
    "copy:posts": "cp -r ../content/posts/*.md src/posts",
    "start": "pnpm run \"/^start:.*/\"",
    "start:content": "pnpm watch",
    "start:11ty": "eleventy --input=src --config=configs/11ty/11ty.config.js --serve --quiet",
    "watch": "npm-watch"
  },
 // ...
```

See example in my blog: https://github.com/noveogroup-amorgunov/amorgunov.com/blob/master/packages/app/package.json#L15

**Comment by zachleat** [comment]
Moderately related to #4021 #3340
